### ISSUE TITLE
how to save the trained model?

### ISSUE THREAD

**Issue by lakezhu** [chelexa/tensorflow-on-android#5]
Hi chelexa,
I call the save_model operation can not create model file,i'm sure the operation returned right value 1.0.

with tf.Session() as sess:
    def save_model():
        path = tf.train.write_graph(sess.graph_def, '/data/data/com.chelexa.tfandroid', 'mnist_50_mlp.pb',as_text=False)
        #path_folder=path[33:36]
        return tf.to_float(tf.rank(path))
    x = tf.placeholder(tf.float32, shape=[None, 784], name="x")
    y = tf.placeholder(tf.float32, [None, 10], name="y")

    W = tf.Variable(tf.zeros([784, 10]), name="weights")
    b = tf.Variable(tf.zeros([10]))

    y_out = tf.matmul(x, W) + b

    #cross_entropy = tf.reduce_mean(-tf.reduce_sum(y * tf.log(y_out), reduction_indices=[1]))
    cross_entropy = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(labels=y, logits=y_out))
    train_step = tf.train.GradientDescentOptimizer(0.5).minimize(cross_entropy, name="train")

    correct_prediction = tf.equal(tf.argmax(y_out,1), tf.argmax(y,1))
    accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32), name="test")

    init = tf.variables_initializer(tf.global_variables(), name="init")
    save = tf.add(1.0,save_model(),name="save_model")
    save_save = tf.add(1.0,save_model(),name="savesave")

thanks!

**Comment by tylermzeller** [comment]
We didn't get that far. This project was an extension of [another project](https://github.com/jihunhamm/Crowd-ML) where multiple devices would download the current model parameters from a server, compute the gradients, and send those gradients back to the server for weight update. So really, a server is supposed to "save" the model.

This project is really just a proof-of-concept to show that we could get tensorflow to train (not just predict) on mobile devices. 

We aren't very active on this repo anymore because of other commitments. That said, if you happen to develop a solution, feel free to submit a pull request. Others have asked about this in the past and may find it useful.
